packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -63,6 +63,7 @@ export function MediaPlaceholder( {
 	isAppender,
 	accept,
 	addToGallery,
+	isGallery = false,
```

Comment:
Does this isGallery need to be different than "multiple = true"?

Reply:
While we are supporting both galleries it does. The isGallery is set by the new gallery code only, and is needed as we need to return early without running the mediaUpload as this is being handled by the nested Image blocks.

The existing gallery still uses the multiple flag still as the media placeholder still still handles some of its file uploading.

We can tidy this up to a single flag again once we remove the old gallery code.

Reply:
> we need to return early without running the mediaUpload as this is being handled by the nested Image blocks.

I'm not sure I understand why we do this.  In theory, we could just leave the MediaPlaceholder as is and create innerBlocks after the upload is complete right?

Reply:
The only thing is we want placeholders in place while the uploads are completing, and the easiest way to get the placeholders in a gallery layout is to create the innerBlocks straight away rather than waiting for uploads to complete. We played around with a few options here and letting the InnerBlocks handle there own upload completion seemed like the simplest solution and gave the least janky experience.

Reply:
I understand but I don't like the fact that we're hijacking a reusable component to do so. I think instead of "isGallery" which provides no meaning for me, maybe it's better if it's the prop is named more something like `shouldUpload` or something like that and that it should work the same regardless of  whether  `multiple` is true or false.

Reply:
Thanks for clarifying, I see where you are coming from now. I have changed the flag attribute to handleUploads and the change to the media placeholder is minimal now.